### PR TITLE
Merges changes from `0.1` to `0.3`

### PR DESCRIPTION
**Pull request by shawyu-okta**
## Description:

`0.3` is branched from `master`
We then want to merge `0.1` -> `0.3`
This results in `0.3` being the new development branch that brings in most of `master` combined with the `0.1` development thus far

## PR Checklist

- [ ] Have you verified the basic functionality for this change?
- [ ] Did you add tests, as appropriate, following our [Automated Test guidelines](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/2676497890/Automated+Testing+in+the+Signin+Widget)?
- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?
- [ ] Did you verify the change by running [downstream monolith artifact](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/102897979/Sign-in+Widget+Development#Sign-inWidgetDevelopment-Instructionstocreateandrunthedownstreamartifact(d16t))? (Provide link to build below)
- [ ] Does this PR include noticeable changes to the UI? (If yes, attach screenshots/video below)

### Issue:

- [OKTA-541757](https://oktainc.atlassian.net/browse/OKTA-541757)

### CODE REVIEW

File: src/util/Bundles.ts [hunk 1]
```diff
@@ -10,12 +10,15 @@
...
 
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-nocheck
+
 import config from 'config/config.json';
 import fetch from 'cross-fetch';
 import country from 'nls/country.json';
 import login from 'nls/login.json';
 import Q from 'q';
-import { _ } from '@okta/courage';
```
Comment (shawyu-okta) [inline src/util/Bundles.ts:18]: todo: need more investigation

File: src/v2/view-builder/internals/BaseForm.ts [hunk 2]
```diff
@@ -84,7 +84,9 @@ export default Form.extend({
      * Rerender would clear infoContainer or views classes can clear it explicitly.
      */
     const infoContainer= '<div class=\'o-form-info-container\'></div>';
-    this.$el.find('.o-form-error-container').before(infoContainer);
+    this.$el.find('.o-form-error-container')
```
Comment (lkchoi) [inline src/v2/view-builder/internals/BaseForm.ts:87]: Getting "Property 'omit' does not exist on type 'OktaUnderscore'.ts(2339)" on L134. Seems unrelated to this change?

**Review by lkchoi — COMMENTED** [review]
Review part 1

File: src/v3/test/integration/authenticator-apple-sso-extension.test.tsx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import identify from '@okta/mocks/data/idp/idx/identify.json';
+import withCredentialMock from '@okta/mocks/data/idp/idx/identify-with-apple-credential-sso-extension.json';
+import noCredentialMock from '@okta/mocks/data/idp/idx/identify-with-no-sso-extension.json';
+import stepUpError from '@okta/mocks/data/idp/idx/error-401-okta-verify-apple-sso-step-up.json';
+import { setup } from './util';
+
+const verifyUrl = 'http://localhost:3000/idp/idx/authenticators/sso_extension/transactions/123/verify?\
```
Comment (lkchoi) [inline src/v3/test/integration/authenticator-apple-sso-extension.test.tsx:19]: Redact?

File: package.json [hunk 4]
```diff
@@ -149,7 +154,7 @@
     "eslint": "^7.23.0",
     "eslint-import-resolver-typescript": "^3.5.0",
     "eslint-import-resolver-webpack": "^0.13.2",
-    "eslint-plugin-compat": "^4.0.2",
+    "eslint-plugin-compat": "4.0.2",
```
Comment (lkchoi) [inline package.json:157]: intentional?

File: .gitignore [hunk 5]
```diff
@@ -25,6 +25,8 @@ test-reports/
 package-lock.json
 .eslintcache
 /types
+types/
```
Comment (lkchoi) [inline .gitignore:28]: Different than `/types` on line above? I think `types/` causes a problem if we create a dir called `types/` to hold our manual type definitions.

File: .eslintignore [hunk 6]
```diff
@@ -14,3 +14,5 @@ types/.eslintrc.js
 types/generated
 types/**/*.test-d.ts
 scripts/dockolith
+
+src/v3/
```
Comment (lkchoi) [inline .eslintignore:18]: This is because we bring our own linting, right? Don't want to double-report or fail based on G2 lint configs

File: .bacon.yml [hunk 7]
```diff
@@ -1,35 +1,66 @@
...
+      sort_order: '5'
+      timeout: '15'
       script_name: publish
       criteria: MERGE
       queue_name: small
     - name: verify-package
       script_path: /root/okta/okta-signin-widget/scripts
-      sort_order: '2'
-      timeout: '10'
+      sort_order: '6'
```
Comment (lkchoi) [inline .bacon.yml:42]: nit: duplicate `sort_order`

**Review by lkchoi — COMMENTED** [review]
review part 2

File: .eslintignore (hunk 6 above)
Comment (shawyu-okta) [reply .eslintignore:18]: Yeah that's right

File: .gitignore (hunk 5 above)
Comment (shawyu-okta) [reply .gitignore:28]: I wasn't sure, probably is duplicate so I'll remove this one.

File: package.json (hunk 4 above)
Comment (shawyu-okta) [reply package.json:157]: Yeah, had some weird issues and needed to pin. We can unpin in separate work later after more investigation if needed

File: .vscode/launch.json [hunk 8]
```diff
@@ -42,39 +42,40 @@
...
       "console": "integratedTerminal",
-      "cwd": "${workspaceRoot}/test/e2e",
-      "env": {
-        "DEBUG": "1"
-      }
+      "cwd": "${workspaceRoot}"
     },
     {
       "type": "node",
+      "protocol": "inspector",
```
[the remark this answers is not in the document]
Comment (shawyu-okta) [reply .vscode/launch.json:66]: looks like a bad merge, will restore from master

File: src/v3/test/integration/authenticator-apple-sso-extension.test.tsx (hunk 3 above)
Comment (shawyu-okta) [reply src/v3/test/integration/authenticator-apple-sso-extension.test.tsx:19]: The actual challengeRequest value is material to the flow working. (I believe it's an encoded JSON string but doesn't seem to be b64)

File: scripts/buildtools/commands/prepack.js [hunk 9]
```diff
@@ -11,6 +11,8 @@ exports.describe = 'Prepares the dist directory for publishing on npm';
 exports.handler = async () => {
   shell.echo(chalk.cyan('Prepacking...'));
 
+  shell.echo('Modifying final package.json');
```
[the remark this answers is not in the document]
Comment (shawyu-okta) [reply scripts/buildtools/commands/prepack.js:14]: I'll remove

File: .testcaferc-parity.js [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+  * test/testcafe/spec/Smoke_spec.js
+  */
+  filter: (testName, fixtureName, fixturePath, testMeta, fixtureMeta) => {
+    return fixtureMeta.v3 === true && testMeta.v3 !== false;
+  },
+  userVariables: {
+    v3: true,
+  },
+  // OKTA-575629 Remove this when v3 parity test flakiness is resolved
+  assertionTimeout: 20000,
```
Comment (leemchale-okta) [inline .testcaferc-parity.js:17]: we can probably remove this and resolve the ticket since its not flakey anymore

Comment (shawyu-okta) [reply .testcaferc-parity.js:17]: I'm going to keep it for now if that's okay, we can remove in a separate PR from the linked JIRA. I want to keep the changes in this merge minimal to the required edits to pass the build and get it merged.

**Review by leemchale-okta — APPROVED** [review]

**Review by glenfannin-okta — APPROVED** [review]

**Review by nikhilvenkatraman-okta — APPROVED** [review]
